### PR TITLE
[crash] Support extra merp params

### PR DESCRIPTION
**Pull request by alexanderkyte**
Fixes https://github.com/mono/mono/issues/11579

### CODE REVIEW

File: mono/utils/mono-merp.c [hunk 1]
```diff
@@ -471,6 +488,15 @@ mono_merp_invoke (const intptr_t crashed_pid, const char *signal, const char *no
 	return success;
 }
 
+void
+mono_merp_add_annotation (const char *key, const char *value)
+{
+	MonoMerpAnnotationEntry *entry = g_new0 (MonoMerpAnnotationEntry, 1);
+	entry->key = g_strdup (key);
+	entry->value = g_strdup (value);
+	config.annotations = g_list_append (config.annotations, entry);
```
Comment (lambdageek) [inline mono/utils/mono-merp.c]: How many annotations are you expecting?  This is an O(n) insertion.  Could use a `GQueue` to get constant-time appends.

Comment (alexanderkyte) [reply mono/utils/mono-merp.c]: Good catch. I think I'll just prepend to an slist.

Comment (alexanderkyte) [reply mono/utils/mono-merp.c]: A queue to me implies that we care about the FIFO behavior. This is going to be immutable after it's installed.

**Comment by alexanderkyte** [comment]
@monojenkins build deb with monolite

**Comment by alexanderkyte** [comment]
@monojenkins build failed

**Comment by alexanderkyte** [comment]
Ah, corlib version was bumped.

**Comment by alexanderkyte** [comment]
@monojenkins build deb with monolite

**Comment by alexanderkyte** [comment]
@monojenkins build failed

**Comment by luhenry** [comment]
@monojenkins backport to 2018-08
@monojenkins backport to 2018-10
@monojenkins backport to 2018-12

**Pull request merged**